ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7316,7 +7316,7 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
       destTableId++;
       // Create the work for moving the table
       // NOTE: specify Dynamic partitions in dest_tab for WriteEntity
-      if (!isNonNativeTable) {
+      if (!isNonNativeTable || destinationTable.getStorageHandler().useNativeCommit()) {
```

Comment:
Can you give some context why this is needed? Thanks!

Reply:
Otherwise we will end up with `PreInsertTableOperation` instead of `MoveTask`.

Reply:
Oh I see. Thanks